test/test_jit.py

```diff
@@ -612,6 +612,22 @@ def forward(self, input):
         m_dropout.eval()
         self.assertEqual(dropout(input) + 1, m_dropout(input))
 
+    def test_script_grad_mode(self):
+        @torch.jit.script
+        def test_set_grad_mode_function(x):
+            # type: (Tensor) -> Tuple[Tensor, Tensor]
+            torch.set_grad_enabled(True)
+            y = x * 2
+            torch.set_grad_enabled(False)
```

Comment:
This is going to leave gradients off after this test, right? That seems like it will break other tests.

Reply:
hmm yeah you are right, there's no API to get the current grad_mode so it's hard to restore the environment as before. We should actually enable the context manager version first instead of acting like a function. I will postpone adding this until we have context manager supported.